go/test/endtoend/docker/vttestserver_test.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	"vitess.io/vitess/go/sqltypes"
+
+	"vitess.io/vitess/go/mysql"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMain(m *testing.M) {
+	exitCode := func() int {
+		err := makeVttestserverDockerImages()
+		if err != nil {
+			return 1
+		}
+		return m.Run()
+	}()
+	os.Exit(exitCode)
+}
+
+func TestUnsharded(t *testing.T) {
+	dockerImages := []string{vttestserverMysql57image, vttestserverMysql80image}
+	for _, image := range dockerImages {
+		t.Run(image, func(t *testing.T) {
+			vtest := newVttestserver(image, []string{"unsharded_ks"}, []int{1}, 1000, 33577)
+			err := vtest.startDockerImage()
+			require.NoError(t, err)
+			defer vtest.teardown()
+
+			// wait for the docker to be setup
+			time.Sleep(10 * time.Second)
```

Comment:
This may end up biting us at some point; IMO it's fine to merge as-is and iterate later, but the bash equivalent of what we can use to check the status for sure would be `docker inspect vttestserver-end2end-test | jq '.[0].State.Health'`; possible values, as far as I know, are `"starting"`, `"healthy"`, and `"unhealthy"`

Reply:
Done in the last commit.

Reply:
Awesome :)